R/preprocessing.R

```diff
@@ -0,0 +1,152 @@
+##'   interpreted as missing and skipped in the likelihood. If set to
+##'   "accumulate", modelled observations will be accumulated and added to the
+##'   next non-missing data point. This can be used to model incidence data that
+##'   is reported at less than daily intervals. If set to "accumulate", the
+##'   first data point is not included in the likelihood (unless `initial` is
+##'   set to a non-zero value) but used only to reset modelled observations to
+##'   zero. If "zero" then all observations on missing dates will be assumed to
+##'   be of value 0.
+##' @param na Character. How to process dates that have NA values. The options
+##'   available are the same ones as for the `missing` argument.
+##' @param initial_accumulate Integer. The number of initial dates to accumulate
+##'   if `missing` or a column name is set to `"accumulate"`. This number of
+##'   dates is added to the beginning of the data set to be accumulated onto the
+##'   first data point. This is useful, for example, for modelling weekly
+##'   incidence data, in which case this should be set to 7. If accumulating and
+##'   the first data point is one to accumulate and this is not set, then that
+##'   data point will be removed with a warning.
+##' @param obs_column Character. If given, only the column specified here will
+##'   be used for checking missingness. This is useful if using a data set that
+##'   has NA values in multiple columns, but only one of them corresponds to
+##'   observations that are to be processed here. By default, all columns will
+##'   be processed and, e.g. `accumulate` will be set if any column is NA in any
+##'   given row.
+##' @return a data.table with an `accumulate` column that indicates whether
+##'   values are accumulated (see the documentation of the `data` argument in
+##'   [estimate_infections()])
+##' @importFrom rlang arg_match
+##' @importFrom data.table as.data.table data.table
+##' @export
+##' @examples
+##' cases <- data.table::copy(example_confirmed)
+##' ## calculate weekly sum
+##' cases[, confirm := data.table::frollsum(confirm, 7)]
+##' ## limit to dates once a week
+##' cases <- cases[seq(7, nrow(cases), 7)]
+##' ## set the second observation to missing
+##' cases[2, confirm := NA]
+##' ## fill missing data
+##' fill_missing(cases, missing = "accumulate", initial_accumulate = 7)
+fill_missing <- function(data,
+                         missing = c("missing", "accumulate", "zero"),
+                         na = c("missing", "accumulate", "zero"),
+                         initial_accumulate,
+                         obs_column) {
+  assert_data_frame(data)
+  assert_names(names(data), must.include = "date", disjunct.from = "accumulate")
+  if (!missing(obs_column)) {
+    assert_names(names(data), must.include = obs_column)
+  } else {
+    obs_column <- setdiff(colnames(data), "date")
+  }
+  assert_date(data$date, any.missing = FALSE)
+
```

Comment:
Need to make sure `initial_accumulate` is never below 0.
```suggestion
assert_character(missing)
assert_character(na)
assert_character(obs_column)
assert_integer(initial_accumulate, lower = 0)
```